youtube_dl/downloader/external.py

```diff
@@ -181,7 +181,8 @@ def _call_downloader(self, tmpfilename, info_dict):
 
         args = [ffpp.executable, '-y']
 
-        args += self._configuration_args()
+        if self.exe == self.get_basename():
```

Comment:
This looks like a hack. We should not actually even allow such situations when they are not equal since it's clearly an inconsistency.
Instead we should ignore explicit `--external-downloader` and `--external-downloader-args` when [selecting different downloader by our own](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/downloader/__init__.py#L44-L50). This can be achieved for example by introducing some temporary fields in `info_dict` for *selected-external-downloader* and it's *args* that should be equal to `--external-downloader` and `--external-downloader-args` when we end up selecting [suggested external downloader](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/downloader/__init__.py#L42) and should not [otherwise](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/downloader/__init__.py#L44-L50). Then in `ExternalFD` these temporary fields should be referenced instead of `params.get('external_downloader')` and `params.get('external_downloader_args')`.